### PR TITLE
Adding Availability components to client and server side

### PR DESCRIPTION
**Pull request by amanmibra**
## Description
This adds the Availability components to the frontend and adds column to each user in the database that holds this information. Also, this PR will allow users to make changes to their availability data from the client side to the server side. This does **NOT** involve the algorithm that is involved to calculate who is available when creating shifts.

## Motivation and Context
This is the the foundation of the full functionality of the availability feature. It is necessary that the availability data can be accessed client side from the server through GET, PUT, and POST requests.

## How Has This Been Tested?
1. Create user with sign up prompts, and see and fill out the new addition availability step
2. Once signed up, go to User Profile and click "Edit User Availability".
3. See how availability is the same as the information you inputted when signed up.
4. Change availability and then click "Save".
5. Refresh page and see updates reflecting on new availability data.

## Screenshots (if appropriate):

image

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by amanmibra** [comment]
Ready for review @vinitbytheminute  @mafuvadze  @Abdulla-Shahid  @rikkikendall

File: app/javascript/src/containers/App.jsx [hunk 1]
```diff
@@ -42,7 +40,6 @@ class App extends Component {
         <ConnectedRouter history={history}>
           <ConnectedSwitch>
             <Route exact path="/" render={() => {
-              console.log('user.isLoggedIn', user.isLoggedIn);
```
Comment (vinitbytheminute) [inline app/javascript/src/containers/App.jsx]: Please remove console.log

**Comment by vinitbytheminute** [comment]
**Review** 
I was unable to find any issues with your PR. The Log-In Availability component worked well and was reflected properly under User Profile/Edit User Availability. The changes in my availability were properly saved and the banner signifying changes was successfully changing was shown.  Good job with your PR!!

**Review by vinitbytheminute — COMMENTED** [review]
Great Job

**Pull request merged**